storage: TestEagerReplication failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/b185d50b579da038cb2a6dd0d0c7ff6088bd00d9

Parameters:
```
COCKROACH_PROPOSER_EVALUATED_KV=true
TAGS=
GOFLAGS=
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=213803&tab=buildLog

```
I170410 06:30:04.652963 51359 gossip/gossip.go:293  [n1] NodeDescriptor set to node_id:1 address:<network_field:"" address_field:"" > attrs:<> locality:<> 
I170410 06:30:04.655402 51359 storage/store.go:1329  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170410 06:30:04.656597 51359 storage/split_queue.go:92  [split,s1,r1/1:/M{in-ax},@c421682000] splitting at key /Table/0/0
I170410 06:30:04.657452 51359 storage/replica_command.go:2701  [split,s1,r1/1:/M{in-ax},@c421682000] initiating a split of this range at key /Table/0 [r2]
I170410 06:30:04.659553 51359 storage/split_queue.go:109  [split,s1,r1/1:/{Min-Table/0},@c421682000] splitting size=246 max=0
I170410 06:30:04.659616 51359 storage/replica_command.go:2701  [split,s1,r1/1:/{Min-Table/0},@c421682000] initiating a split of this range at key /Meta2/Table/0 [r3]
I170410 06:30:04.662171 51311 storage/split_queue.go:92  [split,s1,r2/1:/{Table/0-Max},@c421ee0600] splitting at key /Table/11/0
I170410 06:30:04.662203 51311 storage/replica_command.go:2701  [split,s1,r2/1:/{Table/0-Max},@c421ee0600] initiating a split of this range at key /Table/11 [r4]
W170410 06:30:04.663022 51311 storage/stores.go:218  range not contained in one range: [/Meta2/Table/11,/Table/11/NULL), but have [/Meta2/Table/0,/Table/0)
I170410 06:30:04.668457 51311 storage/split_queue.go:92  [split,s1,r4/1:/{Table/11-Max},@c421fae000] splitting at key /Table/12/0
I170410 06:30:04.668531 51311 storage/replica_command.go:2701  [split,s1,r4/1:/{Table/11-Max},@c421fae000] initiating a split of this range at key /Table/12 [r5]
I170410 06:30:04.672688 51311 storage/split_queue.go:92  [split,s1,r5/1:/{Table/12-Max},@c421ee0300] splitting at key /Table/13/0
I170410 06:30:04.672719 51311 storage/replica_command.go:2701  [split,s1,r5/1:/{Table/12-Max},@c421ee0300] initiating a split of this range at key /Table/13 [r6]
I170410 06:30:04.675352 51311 storage/split_queue.go:92  [split,s1,r6/1:/{Table/13-Max},@c421682600] splitting at key /Table/14/0
I170410 06:30:04.675383 51311 storage/replica_command.go:2701  [split,s1,r6/1:/{Table/13-Max},@c421682600] initiating a split of this range at key /Table/14 [r7]
	soon.go:43: storage/replicate_test.go:64, condition failed to evaluate within 45s: expected 6 replicas in purgatory, but found 7
```

```
[06:45:36][Step 2/2] --- FAIL: TestEagerReplication (45.18s)
[06:45:36][Step 2/2] 	soon.go:43: storage/replicate_test.go:65, condition failed to evaluate within 45s: expected 9 replicas in purgatory, but found 10
```

I can reproduce this locally. Something is creating a 10th range, but only infrequently. This is likely just test flakiness and not something more serious.

--- a/pkg/storage/split_queue.go
+++ b/pkg/storage/split_queue.go
@@ -102,10 +102,12 @@ func (sq *splitQueue) process(ctx context.Context, r *Replica, sysCfg config.Sys
 		return nil
 	}
 
-	// Next handle case of splitting due to size.
+	// Next handle case of splitting due to size. Note that we don't perform
+	// size-based splitting if maxBytes is 0 (happens in certain test
+	// situations).
 	size := r.GetMVCCStats().Total()
 	maxBytes := r.GetMaxBytes()
-	if float64(size)/float64(maxBytes) > 1 {
+	if maxBytes > 0 && float64(size)/float64(maxBytes) > 1 {
 		log.Infof(ctx, "splitting size=%d max=%d", size, maxBytes)
 		if _, validSplitKey, pErr := r.adminSplitWithDescriptor(
 			ctx,

